logbook-servlet/src/main/java/org/zalando/logbook/servlet/RemoteRequest.java

```diff
@@ -114,7 +114,7 @@ public HttpRequest withBody() throws IOException {
     }
 
     private boolean isFormRequest() {
-        return "application/x-www-form-urlencoded".equals(getContentType());
+        return getContentType() != null && getContentType().startsWith("application/x-www-form-urlencoded");
```

Comment:
That would technically also match e.g. `application/x-www-form-urlencoded-foo`. Maybe we could make use of [`javax.activation.MimeType`](https://docs.oracle.com/javase/8/docs/api/javax/activation/MimeType.html#MimeType-java.lang.String-) to parse the media type and check the primary and sub-type specifically.

Reply:
Something along those lines:

```java
@Nullable final String contentType = getContentType();

if (contentType != null) {
    final MimeType type = new MimeType(contentType);
    return "application".equals(type.getPrimaryType()) &&
            "x-www-form-urlencoded".equals(type.getSubType());
}

return false;
```

Reply:
Can `new MimeType(contentType)` fail?

Reply:
> Can new MimeType(contentType) fail?

Technically yes, in practice I don't think so.

Reply:
`new MimeType(rawData)` throws MimeTypeParseException.
I try to extract `try statement`:
```java
@Nullable final String contentType = getContentType();

 private MimeType getMimeType() {
        if (contentType != null) {
            try {
                return new MimeType(contentType);
            } catch (MimeTypeParseException ignored) {

            }
        }
        return new MimeType();
    }

    private boolean isFormRequest() {
        final MimeType type = getMimeType();
        return "x-www-form-urlencoded".equals(type.getSubType()) &&
                "application".equals(type.getPrimaryType());
    }
```
or not:
```java
private boolean isFormRequest() {
        if (contentType != null) {
            try {
                final MimeType type = new MimeType(contentType);
                return "x-www-form-urlencoded".equals(type.getSubType()) && 
                        "application".equals(type.getPrimaryType());
            } catch (MimeTypeParseException ignored) {
    
            }
        }
        return false;
    }
```
I was doing my best. I don't know both of them is best way. 
Is there any other way?

Finally, thank you for answer. this library saved my times 👍